rust/sample_applications/sample_connector_scaffolding/src/main.rs

```diff
@@ -0,0 +1,514 @@
+
+    // Create the appplication context used by the AIO SDK
+    let application_context = ApplicationContextBuilder::default().build()?;
+
+    // Create the Base Connector to handle device endpoints, assets, and datasets creation, update and deletion notifications plus status reporting.
+    let base_connector = BaseConnector::new(application_context);
+
+    // Get the connector artifacts from the base connector, TODO: Use them as needed
+    let _connector_artifacts = base_connector.connector_artifacts();
+
+    // Create a device endpoint client creation observation
+    let device_endpoint_client_creation_observation =
+        base_connector.create_device_endpoint_client_create_observation();
+
+    // Run the session and the base connector concurrently, ending the application if either end (both should run forever unless there are fatal errors)
+    tokio::select! {
+        () = receive_device_endpoints(device_endpoint_client_creation_observation) => {
+            log::warn!("Connector Application tasks ended");
+            Ok(())
+        },
+        res = base_connector.run() => {
+            match res {
+                Ok(()) => {
+                    log::info!("Base connector run completed successfully");
+                    Ok(())
+                }
+                Err(e) => {
+                    log::error!("Base connector run failed: {e}");
+                    Err(Box::new(e))?
+                }
+            }
+        }
+    }
+}
+
+/// Receives a device endpoint and creates a device handler for it.
+///
+/// # Arguments
+/// * `device_endpoint_client_creation_observation` - The device endpoint client creation observation.
+async fn receive_device_endpoints(
+    mut device_endpoint_client_creation_observation: DeviceEndpointClientCreationObservation,
+) {
+    loop {
+        let device_endpoint_client = device_endpoint_client_creation_observation
+            .recv_notification()
+            .await;
+
+        // The log identifier for the device endpoint is used for logging purposes.
+        let device_endpoint_log_identifier = {
+            let device_endpoint_ref = device_endpoint_client.device_endpoint_ref();
+            format!(
+                "[DE: {}_{}]",
+                device_endpoint_ref.device_name, device_endpoint_ref.inbound_endpoint_name
+            )
+        };
+        log::info!("{device_endpoint_log_identifier} Device endpoint created");
+
+        tokio::task::spawn(device_handler(
+            device_endpoint_log_identifier,
+            device_endpoint_client,
+        ));
+    }
+}
+
+/// Handles the device endpoint and receives the asset creation notifications with which it will
+/// create asset handlers.
+///
+/// # Arguments
+/// * `device_endpoint_log_identifier` - A string identifier for the device endpoint, used for logging.
+/// * `device_endpoint_client` - The device endpoint client.
+async fn device_handler(
+    device_endpoint_log_identifier: String,
```

Comment:
I think we should probably bite the bullet even in the scaffolding and use a `SampleConnectorContext` or something here that contains the log identifier since:

1) When OTEL comes in, you'll need it to be here to show how metrics work
2) Artifacts can be shared on it, even if they're unused in this scaffolding, the option is there
3) It gets passed around in exactly the same way as this log identifier, so might as well just use the more scalable pattern

Reply:
agree - I'm going to leave that as a follow-up PR for the sake of time today though

Reply:
sync'd offline, won't make this change at least until OTEL is integrated